tchannel/tchannel.py

```diff
@@ -206,7 +206,7 @@ def decorator(fn):
             return decorator(handler)
 
     @gen.coroutine
-    def advertise(self, routers, name=None, timeout=None):
+    def advertise(self, routers, name=None, timeout=None, router_file=None):
```

Comment:
You need to give `routers` a default value as well.

The way to use this API should not be `tchannel.advertise(None, router_file="foo")`.